### PR TITLE
Update Error Handler Modal UI

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR enhances error handling in the dashboard by improving error message display and adding copy functionality. 

### Detailed summary
- Improved error message display in a modal dialog
- Added copy functionality for error messages
- Refactored error handling logic for better user experience

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4017?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4017** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4017?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4016** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4016?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/@/components/ui/dialog.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ const DialogOverlay = React.forwardRef<
   <DialogPrimitive.Overlay
     ref={ref}
     className={cn(
-      "fixed inset-0 bg-black/80  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",
+      "z-[10000] fixed inset-0 bg-black/80  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/ui/dialog.tsx:24]: last time we did z-index stuff like this it caused a bunch of downstream weirdness are you *sure* this is OK to do?

File: apps/dashboard/src/@/components/ui/dialog.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ const DialogContent = React.forwardRef<
     <DialogPrimitive.Content
       ref={ref}
       className={cn(
-        "fixed grid w-full max-w-lg gap-4 border bg-background p-6 shadow-lg duration-300 rounded-lg border-border",
+        "fixed grid w-full max-w-lg gap-4 border bg-background p-6 shadow-lg duration-300 rounded-lg border-border z-[10001]",
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/ui/dialog.tsx:44]: same z-index concern

File: apps/dashboard/src/components/cmd-k-search/index.tsx [hunk 3]
```diff
@@ -226,8 +226,8 @@ export const CmdKSearch: React.FC = () => {
       {/* modal below here */}
 
       <DialogContent
-        className="z-[10000001] p-0 gap-0"
-        dialogOverlayClassName="z-[10000000] backdrop-blur-sm"
+        className="p-0 gap-0"
+        dialogOverlayClassName="backdrop-blur-sm"
```
Comment (jnsdls) [inline apps/dashboard/src/components/cmd-k-search/index.tsx:230]: last time I asked @jakubkrehel about "hey should we use backdrop-blur` he mentioned we likely shouldn't - can u check with him on that here? (I know we already have it but might as well)

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/@/components/ui/dialog.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/ui/dialog.tsx:24]: removed - moved it to a specific instance

File: apps/dashboard/src/@/components/ui/dialog.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/@/components/ui/dialog.tsx:44]: Removed - moved it to specific instance

File: apps/dashboard/src/components/cmd-k-search/index.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/components/cmd-k-search/index.tsx:230]: Removed

**Review by jnsdls — APPROVED** [review]

**Pull request merged**